pkg/sdkserver/sdkserver.go

```diff
@@ -1077,18 +1085,36 @@ func (s *SDKServer) UpdateList(ctx context.Context, in *beta.UpdateListRequest)
 
 	s.gsUpdateMutex.Lock()
 	defer s.gsUpdateMutex.Unlock()
+	gsCopy := gs.DeepCopy()
 
 	if in.List.Capacity < 0 || in.List.Capacity > apiserver.ListMaxCapacity {
 		return nil, errors.Errorf("out of range. Capacity must be within range [0,1000]. Found Capacity: %d", in.List.Capacity)
 	}
 
-	if _, ok := gs.Status.Lists[name]; ok {
-		batchList := s.gsListUpdates[name]
-		batchList.capacitySet = &in.List.Capacity
-		s.gsListUpdates[name] = batchList
-		// Queue up the Update for later batch processing by updateLists.
-		s.workerqueue.Enqueue(cache.ExplicitKey(updateLists))
-		return &beta.List{}, nil
+	if list, ok := gsCopy.Status.Lists[name]; ok {
+		//capacity := list.Capacity
+		//s.logger.WithField("name", capacity).Debug("Update List -- Currently only used for Updating Capacity")
+
+		// Create *beta.List from *sdk.GameServer_Status_ListStatus for merging.
+		tmpList := &beta.List{Name: name, Capacity: list.Capacity, Values: list.Values}
+		// Removes any fields from the request object that are not included in the FieldMask Paths.
+		fmutils.Filter(in.List, in.UpdateMask.Paths)
+		// Removes any fields from the existing gameserver object that are included in the FieldMask Paths.
+		fmutils.Prune(tmpList, in.UpdateMask.Paths)
+		// Due due filtering and pruning all gameserver object field(s) contained in the FieldMask are overwritten by the request object field(s).
+		proto.Merge(tmpList, in.List)
+		// Silently truncate list values if Capacity < len(Values)
+		if tmpList.Capacity < int64(len(tmpList.Values)) {
+			tmpList.Values = append([]string{}, tmpList.Values[:tmpList.Capacity]...)
+		}
+
+		list.Capacity = tmpList.Capacity
+		list.Values = tmpList.Values
+		gsCopy.Status.Lists[name] = list
+		s.patchGameServer(ctx, gs, gsCopy)
```

Comment:
We don't want to call this here, all the calls are batched, and then the actual change to the gameserver happens in `updatelist` https://github.com/googleforgames/agones/blob/e35d73fcae1c9e7f719d738f4f1638c7ce8d6c7d/pkg/sdkserver/sdkserver.go#L1218

@markmandel do you want to weigh in here? It does seem a bit odd to batch an "overwrite" function?

Reply:
+1 on batching - for a few reasons.

Reply:
+1 to batching for a few reasons: 

1. It puts work items in the workerqueue - which for an "overwrite" operation isn't necessary for conflicts, but is necessary if the KCP goes down for any reason -- so we stay self-healing.
2. The experience across the SDK stays the same regardless of operation.